File: src/tools.js
import { embedText } from "./embeddings.js";
import { searchDocs } from "./qdrant.js";
import { queryLLM } from "./llm.js";

export async function searchTests(query) {
  const vector = await embedText(query);
  return searchDocs(vector, 5);
}

export async function generateFeature(userPrompt) {
  const context = `
You are a Cypress + Cucumber test assistant.
Write ONLY a valid Gherkin .feature file.
Do not add explanations or file paths.
Use this format:

Feature: <name>

  Scenario: <name>
    Given ...
    When ...
    Then ...

`;
  const prompt = `${context}\nUser request: ${userPrompt}`;
  return queryLLM(prompt);
}

File: server/tools.py
from .qdrant_client import get_qdrant_client
from .llm_client import query_llm

def search_tests(query, collection="cypress_tests"):
    client = get_qdrant_client()
    results = client.search(
        collection_name=collection,
        query_vector=query, limit=5
    )
    return results

def generate_new_feature(user_prompt: str):
    context = "You are a Cypress+Cucumber assistant..."
    prompt = f"{context}\nUser request: {user_prompt}"
    return query_llm(prompt)
